lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1192,22 +1190,8 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
                                   AlignInBits, Flags, MangledName);
     }
 
-    case TypeKind::InOut: {
-      // This is an inout type. Naturally we would be emitting them as
-      // DW_TAG_reference_type types, but LLDB can deal better with
-      // pointer-sized struct that has the appropriate mangled name.
-      auto ObjectTy = BaseTy->castTo<InOutType>()->getObjectType();
-      auto Name = MangledName;
-      if (auto *Decl = ObjectTy->getAnyNominal())
-        Name = Decl->getName().str();
-      if (Opts.DebugInfoLevel > IRGenDebugInfoLevel::ASTTypes) {
-        auto DT = getOrCreateDesugaredType(ObjectTy, DbgTy);
-        return createPointerSizedStruct(Scope, Name, DT, File, 0, Flags,
-                                        MangledName);
-      } else
-        return createOpaqueStruct(Scope, Name, File, 0, SizeInBits, AlignInBits,
-                                  Flags, MangledName);
-    }
+    case TypeKind::InOut:
```

Comment:
Will the user-facing inout type disappear, too? Or do we still need to mangle inout types so we can preserve the sugar in LLDB (i.e.: should `p result` show (inout Int)result ?)

Reply:
@adrian-prantl I had a quick discussion with @slavapestov yesterday and he suggested we shouldn't print anything at all.

Reply:
In fact, it's been pointed out that `InOut` types are not real types but `InOut` is more a property  of function parameters so if we want to model we should model it as such.

Reply:
I suppose it is not particularly useful information.